src/ui/public/vis/agg_config.js

```diff
@@ -62,10 +62,11 @@ class AggConfig {
     }, 0);
   }
 
-  constructor(vis, opts = {}) {
+  constructor(vis, opts = {}, aggs) {
     this.id = String(opts.id || AggConfig.nextId(vis.aggs));
     this.vis = vis;
     this._indexPattern = vis.indexPattern;
+    this._aggs = aggs || vis.aggs;
```

Comment:
`this._aggs` is not used inside the AggConfig class, we can remove it so we can get rid of having stored references of it's AggConfigs array inside each object.

Reply:
it is, there are some references to `aggConfig (or config)._aggs` we can't remove just yet

Reply:
Will we work towards removing it completely? Because if we should access it from the outside, we should rather not name this like a private variable?